provider/uprops/src/script.rs

```diff
@@ -174,4 +174,106 @@ mod tests {
             ZeroVec::<Script>::alloc_from_slice(&[Script::Hiragana, Script::Katakana])
         );
     }
+
+    #[test]
+    fn test_has_script() {
+        let root_dir = icu_testdata::paths::uprops_toml_root();
+        let provider = ScriptExtensionsPropertyProvider::try_new(&root_dir)
+            .expect("TOML should load successfully");
+
+        let payload: DataPayload<ScriptExtensionsPropertyV1Marker> = provider
+            .load_payload(&DataRequest {
+                resource_path: ResourcePath {
+                    key: key::SCRIPT_EXTENSIONS_V1,
+                    options: ResourceOptions::default(),
+                },
+            })
+            .expect("The data should be valid")
+            .take_payload()
+            .expect("Loading was successful");
+
+        let scx: &ScriptExtensions = &payload.get().data;
+
+        assert!(scx.has_script('𐓐' as u32, Script::Osage));
+        assert!(!scx.has_script('𐓐' as u32, Script::Common));
+        assert!(!scx.has_script('𐓐' as u32, Script::Inherited));
+
+        assert!(scx.has_script('🥳' as u32, Script::Common));
+        assert!(!scx.has_script('🥳' as u32, Script::Inherited));
+
+        assert!(!scx.has_script(0x200D, Script::Common));
+        assert!(scx.has_script(0x200D, Script::Inherited));
+
+        assert!(scx.has_script('௫' as u32, Script::Tamil));
+        assert!(scx.has_script('௫' as u32, Script::Grantha));
+        assert!(!scx.has_script('௫' as u32, Script::Common));
+        assert!(!scx.has_script('௫' as u32, Script::Inherited));
+
+        assert!(scx.has_script(0x11303, Script::Tamil));
+        assert!(scx.has_script(0x11303, Script::Grantha));
+        assert!(!scx.has_script(0x11303, Script::Common));
+        assert!(!scx.has_script(0x11303, Script::Inherited));
+
+        assert!(scx.has_script(0x30A0, Script::Hiragana));
+        assert!(scx.has_script(0x30A0, Script::Katakana));
+        assert!(scx.has_script(0x30A0, Script::Common));
+        assert!(!scx.has_script(0x30A0, Script::Inherited));
+    }
+
+    #[test]
+    fn test_script_ext_ranges() {
+        let root_dir = icu_testdata::paths::uprops_toml_root();
+        let provider = ScriptExtensionsPropertyProvider::try_new(&root_dir)
+            .expect("TOML should load successfully");
+
+        let payload: DataPayload<ScriptExtensionsPropertyV1Marker> = provider
+            .load_payload(&DataRequest {
+                resource_path: ResourcePath {
+                    key: key::SCRIPT_EXTENSIONS_V1,
+                    options: ResourceOptions::default(),
+                },
+            })
+            .expect("The data should be valid")
+            .take_payload()
+            .expect("Loading was successful");
+
+        let scx: &ScriptExtensions = &payload.get().data;
```

Comment:
Make a helper function for loading the scx data? Just to reduce duplicating boilerplate code.

Reply:
When using the data provider, we want to borrow (via `&` to give us a read-only reference) the data payload in the data provider's response object. This is equivalent to saying that we are avoiding allocating / cloning (sort of known as "zero-copy"?).

Rust's rules on lifetimes disallows us from having a function that returns a reference to some value that gets created within its body as a local variable. Once the function returns, its lexical scope is considered "dropped" (and IRL, I guess its stack frame is discarded?), so trying to have a reference to those dropped stack variables would be a dangling pointer, so the lifetime rules disallow it at compile time.

I haven't created the serialization code and connected this to the testdata provider, I can't have cleaner tests like for the user-facing APIs that take the data provider as a parameter ([example](https://unicode-org.github.io/icu4x-docs/doc/icu_properties/sets/fn.get_emoji.html)). In that example, you can see that you can create an "owned" (opposite of "borrowed") data provider instance struct, and since it is owned, you could make that the return value of a function (which also would transfer ownership of that value to the caller).

Anyways, maybe the current boilerplate duplication is fine in a way because it directly tests the data provider without depending on serialization code. But it is redundant.

Reply:
This "boilerplate" will go away once we have the proper SCX public load functions implemented and it is all plumbed through icu4x-datagen.